kratos/tests/test_matrix_interface.py

```diff
@@ -148,6 +150,49 @@ def test_sqaure_list_of_list_construction(self):
             for j in range(D.Size2()):
                 self.assertEqual(D[i,j], i + (j*0.1))
 
+    def test_buffer_protocol(self) -> None:
+        # + ------------ +
+        # |  0  1  2  3  |
+        # |  4  5  6  7  |
+        # |  8  9 10 11  |
+        # | 12 13 14 15  |
+        # | 16 17 18 19  |
+        # + ------------ +
+        numpy_matrix = numpy.arange(20.0, dtype=numpy.float64).reshape((5, 4))
+
+        # Test default casting
+        kratos_matrix = KM.Matrix(numpy_matrix)
+        self.assertEqual(kratos_matrix.Size1(), numpy_matrix.shape[0])
+        self.assertEqual(kratos_matrix.Size2(), numpy_matrix.shape[1])
+        for i_row in range(numpy_matrix.shape[0]):
+            for i_column in range(numpy_matrix.shape[1]):
+                self.assertEqual(kratos_matrix[i_row, i_column], numpy_matrix[i_row, i_column])
+
+        # Test slicing
+        # + ------ +
+        # |  5  6  |
+        # |  9 10  |
+        # | 13 14  |
+        # + ------ +
+        sliced_numpy_matrix = numpy_matrix[1:4,1:3]
+        kratos_matrix = KM.Matrix(sliced_numpy_matrix)
+        self.assertEqual(kratos_matrix.Size1(), 3)
+        self.assertEqual(kratos_matrix.Size2(), 2)
+        self.assertEqual(kratos_matrix[0, 0], 5.0)
+        self.assertEqual(kratos_matrix[1, 0], 9.0)
+        self.assertEqual(kratos_matrix[2, 0], 13.0)
+        self.assertEqual(kratos_matrix[0, 1], 6.0)
+        self.assertEqual(kratos_matrix[1, 1], 10.0)
+        self.assertEqual(kratos_matrix[2, 1], 14.0)

```

Comment:
`self.assertMatrixAlmostEqual`?

Reply:
can't use it unfortunately :( `assertMatrixAlmostEqual` expects `Kratos::Matrix` arguments (for example, `numpy.ndarray` doesn't have `Size1` or `Size2`)